src/AWS.Deploy.Orchestration/DeploymentCommands/BeanstalkEnvironmentDeploymentCommand.cs

```diff
@@ -3,19 +3,66 @@
 
 using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
 using System.Threading.Tasks;
 using AWS.Deploy.Common;
 using AWS.Deploy.Orchestration.DisplayedResources;
 
 namespace AWS.Deploy.Orchestration.DeploymentCommands
 {
-    /// <summary>
-    /// This class is a Work In Progress.
-    /// </summary>
     public class BeanstalkEnvironmentDeploymentCommand : IDeploymentCommand
     {
-        public Task ExecuteAsync(Orchestrator orchestrator, CloudApplication cloudApplication, Recommendation recommendation) => throw new NotImplementedException();
-        public Task<List<DisplayedResourceItem>> GetDeploymentOutputsAsync(IDisplayedResourcesHandler displayedResourcesHandler, CloudApplication cloudApplication, Recommendation recommendation) => throw new NotImplementedException();
+        public async Task ExecuteAsync(Orchestrator orchestrator, CloudApplication cloudApplication, Recommendation recommendation)
+        {
+            if (orchestrator._interactiveService == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._interactiveService)} is null as part of the orchestartor object");
+            if (orchestrator._awsResourceQueryer == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._awsResourceQueryer)} is null as part of the orchestartor object");
+            if (orchestrator._elasticBeanstalkHandler == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._elasticBeanstalkHandler)} is null as part of the orchestartor object");
+            if (orchestrator._s3Handler == null)
+                throw new InvalidOperationException($"{nameof(orchestrator._s3Handler)} is null as part of the orchestartor object");
+
```

Comment:
why are we accessing a lot of the services through the orchestrator and not using dependency injection on this class and accessing them directly?

Reply:
All classes them implement the `IDeploymentCommand` interface are instantiated during runtime via the [DeploymentCommandFactory](https://github.com/aws/aws-dotnet-deploy/blob/kmalhar/beanstalk-backwards-compatibility/src/AWS.Deploy.Orchestration/DeploymentCommands/DeploymentCommandFactory.cs#L28). It is instantiated with a parameterless constructor and thus these classes cannot have any dependencies.

I agree that attaching each service handler in the `Orchestrator` will make it bloated. To handle this I created a single `AWSServiceHander` class which will be part of the `Orchestrator` and all individual service handlers will be a part of this class via dependency injection.